code/game/objects/items/devices/healthanalyzerpro.dm

```diff
@@ -0,0 +1,425 @@
+/obj/item/device/healthanalyzerpro/
+	name = "ProHealth Analyzer"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/misc_tools.dmi', "right_hand" = 'icons/mob/in-hand/right/misc_tools.dmi')
+	icon = 'icons/obj/device.dmi'
+	icon_state = "adv_health"
+	item_state = "healthanalyzer"
+	desc = "A hand-held body scanner able to precisely distinguish vital signs of the subject. This particular device is an experimental model outfitted with several modules that fulfill the roles of common scanning tools, memory function to record last made scan and a printer."
+	flags = FPRINT
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throwforce = 5
+	w_class = W_CLASS_MEDIUM
+	throw_speed = 4
+	starting_materials = list(MAT_IRON = 700, MAT_PLASTIC = 200, MAT_URANIUM = 50, MAT_SILVER = 50, MAT_GOLD = 50)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = Tc_MAGNETS + "=4;" + Tc_BIOTECH + "=4"
+	attack_delay = 0
+	var/last_scantime = 0
+	var/last_reading = null
+	var/mode = "Health Scan"
+	var/list/modes = list("Health Scan", "Simplified Health Scan", "Advanced Health Scan", "Reagents Scan", "Immunity Scan", "Autopsy Scan")
+	var/obj/item/device/antibody_scanner/immune
+	var/last_print
+	parent_type = /obj/item/weapon/autopsy_scanner
+
+/obj/item/device/healthanalyzerpro/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Current active mode: [mode].</span>")
```

Comment:
info for descriptive class

Reply:
one doesn't need to look much to know what each mode does
it doesn't fire a nuke on any of them too

Reply:
spawn class should be 'info'

Reply:
oh, I see, the description text flavor